src/GSoft.ComponentModel.DataAnnotations.Tests/GuidAttributeTests.cs

```diff
@@ -90,25 +92,43 @@ public void IsValid_Returns_False_When_Value_Is_Not_A_String()
     public void Validator_TryValidateObject_Returns_The_Expected_Error_Messages_When_Validation_Fails()
     {
         var something = new Something();
-        var expectedValue1ErrorMessage = GuidAttribute.ErrorMessageWithoutFormatFormat.FormatInvariant(nameof(Something.Value1));
-        var expectedValue2ErrorMessage = GuidAttribute.ErrorMessageWithFormatFormat.FormatInvariant("D").FormatInvariant(nameof(Something.Value2));
+        var expectedValue1ErrorMessage = GuidAttribute.ErrorMessageWithoutFormatFormat.FormatInvariant(nameof(Something.Valid1), string.Empty);
+        var expectedValue2ErrorMessage = GuidAttribute.ErrorMessageWithFormatFormat.FormatInvariant(nameof(Something.Value2), string.Empty, "D");
+        var expectedValue3ErrorMessage = GuidAttribute.ErrorMessageWithoutFormatFormat.FormatInvariant(nameof(Something.Value3), GuidAttribute.ErrorMessageNonEmptyPart);
+        var expectedValue4ErrorMessage = GuidAttribute.ErrorMessageWithoutFormatFormat.FormatInvariant(nameof(Something.Value4), GuidAttribute.ErrorMessageNonEmptyPart);
 
         var results = new List<ValidationResult>();
         var context = new ValidationContext(something, serviceProvider: null, items: null);
         var isValid = Validator.TryValidateObject(something, context, results, validateAllProperties: true);
 
         Assert.False(isValid);
-        Assert.Equal(2, results.Count);
-        Assert.Single(results, x => x.ErrorMessage == expectedValue1ErrorMessage);
-        Assert.Single(results, x => x.ErrorMessage == expectedValue2ErrorMessage);
+        Assert.Equal(4, results.Count);
+
+        var errorMessages = results.Select(x => x.ErrorMessage).ToArray();
+        Assert.Single(errorMessages, expectedValue1ErrorMessage);
+        Assert.Single(errorMessages, expectedValue2ErrorMessage);
+        Assert.Single(errorMessages, expectedValue3ErrorMessage);
+        Assert.Single(errorMessages, expectedValue4ErrorMessage);
     }
 
     private class Something
```

Comment:
I know it's just used here, but maybe we could give it a more significant name? Something like GuidValues or GuidSamples or other?

Reply:
`Something` is not that great as a class name but it's supposed to represent any class that consumers would like to validate... also would have to change it in other test classes. EDIT: we discussed about that, we'll leave it as-is.